styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -82,11 +92,42 @@ public SchedulerResource(
         Route.with(
             em.response(WorkflowInstance.class),
             "POST", BASE + "/trigger",
-            rc -> this::triggerWorkflowInstance))
+            rc -> this::triggerWorkflowInstance),
+        Route.with(
+            em.response(Workflow.class),
+            "POST", BASE + "/workflows",
+            rc -> this::createOrUpdateWorkflow),
+        Route.with(
+            em.serializerResponse(ByteString.class),
+            "DELETE", BASE + "/workflows/<cid>/<wfid>",
+            rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc)))
+    )
 
         .map(r -> r.withMiddleware(Middleware::syncToAsync));
   }
 
+  private Response<ByteString> deleteWorkflow(String cid, String wfid) {
+    WorkflowId workflowId = WorkflowId.create(cid, wfid);
+    Optional<Workflow> workflowOpt = null;
+    try {
+      workflowOpt = storage.workflow(workflowId);
```

Comment:
We discussed this and decided to do all logic in the SchedulerResource, while WorkflowResource simply functions as a proxy